### PR TITLE
2033 - single case use customer id from case instead of ticket

### PR DESCRIPTION
**Pull request by islean**
## Description

When using rsync, the customer id was generally fetched via the ticket number. In single case instances, we instead use the case given to retrieve the customer id.

### Added

- Test case for moved exception

### Changed

- slurm_rsync_single_case takes a case object instead of case_id as input. 
- slurm_rsync_single_case gets the customer_id from the provided case object.
- get_source_and_destination_paths takes customer_id as input and does not handle logic concerning retrieving that.

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b 2033-rsync-case-customer-change
    ```

### How to test

- [x] Run `cg upload clinical-delivery newhound`
- [x] Run `cg deliver ticket -t 231889 -d fastq -d mip-rna`

### Expected test outcome

- [x] Check that source and destination paths are added to the rsync files when running `cg upload clinical-delivery newhound`.
- [x] Check that `cg deliver ticket -t 231889 -d fastq -d mip-rna` works as previously.

## Review

- [x] Tests executed by VJ and IO
- [x] "Merge and deploy" approved by VJ
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/upload/clinical_delivery.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def upload_clinical_delivery(context: click.Context, case_id: str, dry_run: bool
 
     rsync_api: RsyncAPI = RsyncAPI(context.obj)
     is_complete_delivery, job_id = rsync_api.slurm_rsync_single_case(
-        case_id=case_id,
+        case=case_obj,
```
Comment (ChrOertlin) [inline cg/cli/upload/clinical_delivery.py:59]:
```suggestion
        case=case,
```

in others parts of the code we try to refrain from using siffuxies like _obj but keep it simple. case for singular and cases for plural if it is about the model object

File: cg/meta/rsync/rsync_api.py [hunk 2]
```diff
@@ -100,19 +100,11 @@ def set_log_dir(self, folder_prefix: str) -> None:
...
-            raise CgError()
-        customer_id: str = cases[0].customer.internal_id
-        source_and_destination_paths["delivery_source_path"]: Path = Path(
-            self.delivery_path, customer_id, INBOX_NAME, ticket
-        )
-        source_and_destination_paths["rsync_destination_path"]: Path = Path(
-            self.destination_path, customer_id, INBOX_NAME
-        )
+    def get_source_and_destination_paths(self, ticket: str, customer_id: str) -> Dict[str, Path]:
+        source_and_destination_paths: Dict[str, Path] = {
```
Comment (ChrOertlin) [inline cg/meta/rsync/rsync_api.py:104]:
```suggestion
    def get_source_and_destination_paths(self, ticket: str, customer_id: str) -> Dict[str, Path]:
    """Return the source and destination paths."""
        source_and_destination_paths: Dict[str, Path] = {
```

File: cg/meta/rsync/rsync_api.py [hunk 3]
```diff
@@ -100,19 +100,11 @@ def set_log_dir(self, folder_prefix: str) -> None:
...
-            LOG.warning("Could not find any cases for ticket %s", ticket)
-            raise CgError()
-        customer_id: str = cases[0].customer.internal_id
-        source_and_destination_paths["delivery_source_path"]: Path = Path(
-            self.delivery_path, customer_id, INBOX_NAME, ticket
-        )
-        source_and_destination_paths["rsync_destination_path"]: Path = Path(
-            self.destination_path, customer_id, INBOX_NAME
-        )
+    def get_source_and_destination_paths(self, ticket: str, customer_id: str) -> Dict[str, Path]:
```
Comment (ChrOertlin) [inline cg/meta/rsync/rsync_api.py:103]:
```suggestion
    def get_source_and_destination_paths(self, ticket: str, customer_internal_id: str) -> Dict[str, Path]:
```

File: tests/meta/rsync/test_rsync.py [hunk 4]
```diff
@@ -28,7 +28,9 @@ def test_get_source_and_destination_paths(
     RsyncAPI.get_all_cases_from_ticket.return_value = [case]
 
     # WHEN the source path is created
-    source_and_destination_paths = rsync_api.get_source_and_destination_paths(ticket=ticket_id)
+    source_and_destination_paths = rsync_api.get_source_and_destination_paths(
+        ticket=ticket_id, customer_id=case.customer.internal_id
```
Comment (ChrOertlin) [inline tests/meta/rsync/test_rsync.py:32]:
```suggestion
        ticket=ticket_id, customer_internal_id=case.customer.internal_id
```

File: tests/meta/rsync/test_rsync.py [hunk 5]
```diff
@@ -205,7 +206,7 @@ def test_slurm_rsync_single_case(
     sbatch_number: int
     is_complete_delivery: bool
     is_complete_delivery, sbatch_number = rsync_api.slurm_rsync_single_case(
-        case_id=case_obj.internal_id,
+        case=case_obj,
```
Comment (ChrOertlin) [inline tests/meta/rsync/test_rsync.py:209]:
```suggestion
        case=case,
```

File: tests/meta/rsync/test_rsync.py [hunk 6]
```diff
@@ -245,7 +246,7 @@ def test_slurm_rsync_single_case_missing_file(
...
```
Comment (ChrOertlin) [inline tests/meta/rsync/test_rsync.py:249]:
```suggestion
        case=case,
```

File: tests/meta/rsync/test_rsync.py [hunk 7]
```diff
@@ -43,22 +45,6 @@ def test_get_source_and_destination_paths(
...
-    with pytest.raises(CgError):
-        # WHEN the source path is collected
-        rsync_api.get_source_and_destination_paths(ticket=ticket_id)
-
-        # THEN the source path ends with a customer id, followed by "inbox" and a ticket_id id
-        assert "Could not find any cases for ticket_id" in caplog.text
-
-
 def test_set_log_dir(rsync_api: RsyncAPI, ticket_id: str, caplog):
     """Test function to set log dir for path"""
```
Comment (ChrOertlin) [inline tests/meta/rsync/test_rsync.py:49]:
```suggestion
def test_set_log_dir(rsync_api: RsyncAPI, ticket_id: str, caplog):
    """Test function to set log dir for path."""
```

**Review by ChrOertlin — APPROVED** [review]
Looks good, just some minor things

**Review by henrikstranneheim — APPROVED** [review]
Nice 👍

File: tests/meta/rsync/test_rsync.py (hunk 5 above)
Comment (islean) [reply tests/meta/rsync/test_rsync.py:209]: There are three fixtures used in the test with "_obj". Would it be good for me to change all of these and reconfigure the tests?

**Comment by islean** [comment]
Deployed to stage and production.

**Pull request merged**